### PR TITLE
Override pages guide

### LINKED ISSUE #237: override documents
document steps on how to override:
* records landing page
* records search page
* deposit page

check this PR #238 for the records landing page and deposit page.
check this PR #218 for search page.

The steps to override these pages - should b written in https://github.com/tu-graz-library/docs-repository

### PR DESCRIPTION
**Pull request by rekt-hard**
closes tu-graz-library/invenio-theme-tugraz/issues/237

### CODE REVIEW

File: docs/configs/override-pages.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+  "ResultsList.item": RDMRecordResultsListItem,
+  "SearchApp.facets": RDMRecordFacets,
+  "SearchApp.searchbarContainer": RDMRecordSearchBarContainer,
+  "SearchBar.element": RDMRecordSearchBarElement,
+  "SearchFilters.ToggleComponent": RDMToggleComponent,
+  "Count.element": RDMCountComponent,
+});
+```
+
+#### Adding to webpack
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
#### Update webpack
```

File: docs/configs/override-pages.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
+# Override Pages
+In some cases, the default pages do not offer all the functionality or modification possibilities needed. It is possible to override the pages, though this should be the last resort as it increases resources needed for maintainability.
+
```
Comment (mb-wali) [inline docs/configs/override-pages.md:3]:
```suggestion
We will look into - how to override some essential pages of `invenio-app-rdm` with our existing module `invenio-theme-tugraz`
```

File: docs/configs/override-pages.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+Create new components:
+```js
+export const RDMRecordSearchBarContainer = () => {
+  return (
+    <Overridable id={"SearchApp.searchbar"}>
+      <SearchBar />
+    </Overridable>
+  );
+};
```
Comment (mb-wali) [inline docs/configs/override-pages.md]: this does not make sense - if this belongs to the override of the search page - add it to the section.

**Review by mb-wali — CHANGES_REQUESTED** [review]

File: docs/configs/override-pages.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+
+### Troubleshoot
+
+#### ManifestKeyNotFoundError
+- check if the spelling of the webpack names matches with the import in the template.
+- reinstall invenio-theme-tugraz
+    - In order to pick up all the new files and to rebuild webpack, it is necessary to install `invenio-theme-tugraz` again. If no such changes happened, this step can be skipped.
+    - `invenio-cli packages install /path/to/invenio-theme-tugraz`
+
```
Comment (mb-wali) [inline docs/configs/override-pages.md]: add this at end

File: docs/configs/override-pages.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+
+## Main Search
+The component to be overridden is `search` so the path for this looks like `invenio_theme_tugraz/assets/semantic-ui/js/invenio_theme_tugraz/search/`.
+Place the components in `invenio_theme_tugraz/assets/semantic-ui/js/invenio_theme_tugraz/search/components.js`.
+
+The original file can be found here: [components.js](https://github.com/inveniosoftware/invenio-app-rdm/blob/v6.0.2/invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/search/components.js)
+
+
+#### Creating Search App
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Creating Search App
```

File: docs/configs/override-pages.md [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+  "ResultsList.item": RDMRecordResultsListItem,
+  "SearchApp.facets": RDMRecordFacets,
+  "SearchApp.searchbarContainer": RDMRecordSearchBarContainer,
+  "SearchBar.element": RDMRecordSearchBarElement,
+  "SearchFilters.ToggleComponent": RDMToggleComponent,
+  "Count.element": RDMCountComponent,
+});
+```
+
+#### Update webpack
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Update webpack
```

File: docs/configs/override-pages.md [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+```
+with
+```python
+{%- block javascript %}
+    {{ super() }}
+    {{ webpack['invenio-theme-tugraz-rdm-search.js'] }}
+{%- endblock %}
+```
+
+#### Override Config Variable
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Override Config Variable
```

File: docs/configs/override-pages.md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+  "SearchApp.layout": RDMUserRecordsSearchLayout,
+  "SearchApp.results": RDMDepositResults,
+  "SearchBar.element": RDMRecordSearchBarElement,
+  "BucketAggregation.element": RDMBucketAggregationElement,
+  "BucketAggregationValues.element": RDMRecordFacetsValues,
+  "SearchFilters.ToggleComponent": RDMToggleComponent,
+});
+```
+
+#### Update webpack
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Update webpack
```

File: docs/configs/override-pages.md [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+"semantic-ui": dict(
+            entry={
+                "invenio-theme-tugraz-theme": "./less/invenio_theme_tugraz/theme.less",
+                "invenio-theme-tugraz-js": "./js/invenio_theme_tugraz/theme.js",
+                # add user record search component
+                'invenio-theme-tugraz-rdm-user-records-search': './js/invenio_theme_tugraz/user_records_search/index.js',
+            }, ...
+```
+
+#### Adding Search Template
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding Search Template
```

File: docs/configs/override-pages.md [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+```
+with
+```python
+{%- block javascript %}
+  {{ super() }}
+  {{ webpack['invenio-theme-tugraz-rdm-user-records-search.js'] }}
+{%- endblock javascript %}
+```
+
+#### Adding Render Function
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding Render Function
```

File: docs/configs/override-pages.md [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+@login_required
+def deposit_search():
+    """List of user deposits page."""
+    return render_template(
+        "invenio_theme_tugraz/records/search_deposit.html",
+        searchbar_config=dict(searchUrl=get_search_url()),
+    )
+```
+
+#### Adding URL Route
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding URL Route
```

File: docs/configs/override-pages.md [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+
+## Record Landing Page
+Since we do not need to add additional functionality to the record landing page, there is no need for new components or updating the webpack file. We will simply add and modify the html file and add the URL route. If there is need for additional functionality, follow the steps from the search page guide.
+
+#### Adding Detail Template
+Create the following file `invenio_theme_tugraz/templates/invenio_theme_tugraz/records/detail.html` with the content of the [base record detail template from invenio-app-rdm ](https://github.com/inveniosoftware/invenio-app-rdm/blob/v6.0.2/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html).
+
+Modify the template file as you see fit.
+
+#### Adding Render Function
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding Render Function
```

File: docs/configs/override-pages.md [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+        record=UIJSONSerializer().serialize_object_to_dict(record.to_dict()),
+        pid=pid_value,
+        files=files_dict,
+        permissions=record.has_permissions_to(['edit', 'new_version', 'manage',
+                                               'update_draft', 'read_files']),
+        is_preview=is_preview,
+    )
+```
+
+#### Adding URL Route
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding URL Route
```

File: docs/configs/override-pages.md [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+    record={getInputFromDOM("deposits-record")}
+    files={getInputFromDOM("deposits-record-files")}
+    config={getInputFromDOM("deposits-config")}
+    permissions={getInputFromDOM("deposits-record-permissions")}
+  />,
+  document.getElementById("deposit-form")
+);
+```
+
+#### Update webpack
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Update webpack
```

File: docs/configs/override-pages.md [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+"semantic-ui": dict(
+            entry={
+                "invenio-theme-tugraz-theme": "./less/invenio_theme_tugraz/theme.less",
+                "invenio-theme-tugraz-js": "./js/invenio_theme_tugraz/theme.js",
+                # add deposit component
+                'invenio-theme-tugraz-rdm-deposit': './js/invenio_theme_tugraz/deposit/index.js',
+            }, ...
+```
+
+#### Adding Deposit Template
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding Deposit Template
```

File: docs/configs/override-pages.md [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+        "invenio_theme_tugraz/records/deposit.html",
+        forms_config=get_form_config(apiUrl=f"/api/records/{pid_value}/draft"),
+        record=record,
+        files=draft_files.to_dict(),
+        searchbar_config=dict(searchUrl=get_search_url()),
+        permissions=draft.has_permissions_to(['new_version'])
+    )
+```
+
+#### Adding URL Route
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
### Adding URL Route
```

**Review by mb-wali — CHANGES_REQUESTED** [review]

File: docs/configs/override-pages.md [hunk 17]
```diff
@@ -0,0 +1,365 @@
...
+        files=files_dict,
+        permissions=record.has_permissions_to(['edit', 'new_version', 'manage',
+                                               'update_draft', 'read_files']),
+        is_preview=is_preview,
+    )
+```
+
+#### Adding URL Route
+All that is left is to add a URL rule, so that the new function is called instead of the one from the base implementation. In `invenio_theme_tugraz/ext.py` import the previously defined function and inside the `init_app` function add:
+
```
[the remark this answers is not in the document]
Comment (rekt-hard) [reply docs/configs/override-pages.md:234]: also added for other pages

File: docs/configs/override-pages.md [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+    {{ webpack['invenio-app-rdm-search.js'] }}
+{%- endblock %}
+```
+with
+```python
+{%- block javascript %}
+    {{ super() }}
+    {{ webpack['invenio-theme-tugraz-rdm-search.js'] }}
+{%- endblock %}
+```
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
```diff
{%- block javascript %}
    {{ super() }}
-   {{ webpack['invenio-app-rdm-search.js'] }}
+   {{ webpack['invenio-theme-tugraz-rdm-search.js'] }}
{%- endblock %}
```
```

File: docs/configs/override-pages.md [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+  {{ webpack['invenio-app-rdm-user-records-search.js'] }}
+{%- endblock javascript %}
+```
+with
+```python
+{%- block javascript %}
+  {{ super() }}
+  {{ webpack['invenio-theme-tugraz-rdm-user-records-search.js'] }}
+{%- endblock javascript %}
+```
```
Comment (mb-wali) [inline docs/configs/override-pages.md]:
```suggestion
```diff
{%- block javascript %}
  {{ super() }}
-  {{ webpack['invenio-app-rdm-user-records-search.js'] }}
+  {{ webpack['invenio-theme-tugraz-rdm-user-records-search.js'] }}
{%- endblock javascript %}
```
```

File: docs/configs/override-pages.md [hunk 20]
```diff
@@ -0,0 +1,354 @@
...
+
+### Adding Render Function
+Next, in `invenio_theme_tugraz/deposits.py` we define a function which will render the previously created template and pass the `searchbar_config` as argument.
+```python
+@login_required
```
Comment (mb-wali) [inline docs/configs/override-pages.md:162]: add all required import if there is any.

**Review by mb-wali — CHANGES_REQUESTED** [review]
All looks good! add the changes requsted.

**Review by mb-wali — APPROVED** [review]

**Pull request merged**